bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadata.java

```diff
@@ -185,7 +226,7 @@ private LedgerMetadata() {
         return ensembles;
     }
 
-    void setEnsembles(TreeMap<Long, ? extends List<BookieSocketAddress>> newEnsembles) {
+    void setEnsembles(Map<Long, ? extends List<BookieSocketAddress>> newEnsembles) {
         this.ensembles = newEnsembles.entrySet().stream()
             .collect(TreeMap::new,
```

Comment:
Treemap will be mutable, no?
What about wrapping the result Collections.unmodifiableSortedMap?

Reply:
Eventually. I can't make the data structure immutable yet until the whole code base for metadata is